### ISSUE TITLE
How to embed the xml into the pdf

### ISSUE THREAD

**Issue by flo2000ace** [stephanstapel/ZUGFeRD-csharp#205]
Hello,

I am not from Germany but I want to integrate ZUGFeRD for German Software users, I looked into your Documentation on the First Page and I couldn't find anything related to embedding the XML. Or is this not the goal of this library and it's just for creating the XML? If this is the case could anyone provide an open-source library to do that?

Yours,
Florian

**Comment by stephanstapel** [comment]
Hi Florian,
the goal of the library is to generate (and read) the xml file. Converting a PDF file into PDF/A and embededed the xml file is not the goal.
For extracting the latter from pdf file, there is a short introduction on the readme page:
https://github.com/stephanstapel/ZUGFeRD-csharp (on the bottom)

If you have a similar idea for embedding it, I'd be more than happy to share this as well.

**Comment by flo2000ace** [comment]
Hi Stephan,

I think I found what I was looking for, I hope I can help anyone else with that, also my questions to chatgpt weren't very helpful, but I found this: 
https://stackoverflow.com/questions/70597318/af-reference-to-file-embedded-into-a-pdf-with-itextsharp

I as a foreigner find it very exhausting to include german standards into my software mostly they are covered by individuals who do the most part of the job, so thank you for your work!

**Comment by stephanstapel** [comment]
unfortunately, itextsharp is not free any more afaik.

**Comment by flo2000ace** [comment]
Yes, that's true I also looked up migradoc, but there this is not possible in the way I think it is intended to be, you can add another page and paste the XML invisible there, but I don't think this is a valid approach, or maybe also paste the XML on the invoice page? I dismissed this approach...

Maybe there is a free library in another language i will inform you in this ticket when I find another approach.

**Comment by stephanstapel** [comment]
Thanks!
